src/features/usersSlice.ts

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable no-param-reassign */
+import {
+  ActionReducerMapBuilder, createAsyncThunk, createSlice,
+} from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { getUsers } from '../api/users';
+
+export type UsersState = {
+  users: User[],
+  status: 'idle' | 'loading' | 'failed',
+};
```

Comment:
Don`t use one key status for all response variants. Need to create loaded, hasError etc state key for it.